benchexec/tablegenerator/react-table/src/components/TaskDefinitionViewer.js

```diff
@@ -0,0 +1,114 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * BenchExec is a framework for reliable benchmarking.
+ * This file is part of BenchExec.
+ * Copyright (C) Dirk Beyer. All rights reserved.
+ */
+import React from "react";
+import yamlParser from "yaml";
+
+/** Special view for YAML files in the LinkOverlay component. */
+export default class TaskDefinitionViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      splitterTag: "<splitter#9d81y23>",
+      fileTag: "<file#092nt43>",
+      content: this.props.yamlText,
+    };
+  }
+
+  componentDidMount() {
+    this.prepareTextForRendering();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.yamlText !== this.props.yamlText) {
+      this.prepareTextForRendering();
+    }
+  }
+
+  /**
+   * Parses the YAML file and encloses all input and property files with a fileTag as well as a splitterTag,
+   * so they can be rendered separately as links. Takes the following assumptions:
+   * input_files is either a string or a list of strings
+   * properties is a list of dicts, each with a "property_file" key
+   */
+  prepareTextForRendering = () => {
+    const yamlObj = yamlParser.parseDocument(this.props.yamlText);
+
+    const inputFiles = yamlObj.get("input_files");
+    if (inputFiles) {
+      if (Array.isArray(inputFiles.items)) {
+        inputFiles.items.forEach(inputFileItem => {
+          inputFileItem.value = this.encloseFileInTags(inputFileItem.value);
+        });
+      } else {
+        yamlObj.set("input_files", this.encloseFileInTags(inputFiles));
+      }
+    }
+
+    const properties = yamlObj.get("properties");
+    if (properties) {
+      if (Array.isArray(properties.items)) {
+        properties.items.forEach(property => {
+          if (Array.isArray(property.items)) {
+            property.items.forEach(propertyItem => {
+              if (propertyItem.key.value === "property_file") {
+                propertyItem.value.value = this.encloseFileInTags(
+                  propertyItem.value.value,
+                );
+              }
+            });
+          }
+        });
+      }
+    }
+
+    this.setState({ content: yamlObj.toString() });
+  };
+
+  encloseFileInTags = fileName => {
+    return (
+      this.state.splitterTag +
+      this.state.fileTag +
+      fileName +
+      this.state.fileTag +
+      this.state.splitterTag
+    );
+  };
+
+  render() {
+    const contentBySplitter = this.state.content.split(this.state.splitterTag);
+    const jsxContent = [];
+
+    contentBySplitter.forEach(contentPart => {
```

Comment:
Instead of `forEach` with a mutable `jsxContent`, I suggest to simply use `map`. And `jsxContentPart` could then be replaced with directly returning the result.